pkg/sfu/buffer/wraparound.go

```diff
@@ -0,0 +1,116 @@
+package buffer
+
+type number interface {
+	uint16 | uint32
+}
+
+type extendedNumber interface {
+	uint32 | uint64
+}
+
+type WrapAround[T number, ET extendedNumber] struct {
+	fullRange ET
+
+	initialized bool
+	start       T
+	highest     T
+	cycles      int
+}
+
+func NewWrapAround[T number, ET extendedNumber](fullRange ET) *WrapAround[T, ET] {
```

Comment:
I was checking with @davidzhao about getting this range automatically from type. He mentioned trying to use an interface which can return the value for given type. But, I could not get it to work. So, just passing it in for now.

Reply:
maybe like this?
```go
var t ET
fullRange := 1 << (unsafe.Sizeof(t) * 8)
```

Reply:
oh cool, will try that. Thank you!